### PR TITLE
Drop OPENSSL_NO_RSA everywhere

### PR DESCRIPTION
**Pull request by levitte**
The configuration option 'no-rsa' was dropped with OpenSSL 1.1.0, so
this is simply a cleanup of the remains.

### CODE REVIEW

**Comment by paulidale** [comment]
CIs are relevant.

File: apps/speed.c [hunk 1]
```diff
@@ -155,8 +155,7 @@ static int usertime = 1;
 
 static double Time_F(int s);
 static void print_message(const char *s, long num, int length, int tm);
-#if (!defined(OPENSSL_NO_RSA) && !defined(OPENSSL_NO_DEPRECATED_3_0)) \
-    || (!defined(OPENSSL_NO_DSA) && !defined(OPENSSL_NO_DEPRECATED_3_0)) \
+#if !defined(OPENSSL_NO_DEPRECATED_3_0) || !defined(OPENSSL_NO_DSA) \
```
Comment (paulidale) [inline apps/speed.c:158]: Wouldn't this normally be broken up differently?  Four lines?

File: engines/e_devcrypto.c [hunk 2]
```diff
@@ -1252,9 +1252,7 @@ static int bind_devcrypto(ENGINE *e) {
  * /Richard Levitte, 2017-05-11
  */
 #if 0
```
Comment (paulidale) [inline engines/e_devcrypto.c:1254]: Why not remove the #if 0 block entirely?

File: apps/speed.c (hunk 1 above)
Comment (levitte) [reply apps/speed.c:158]: Uhm, not necessarily?  Considering this is just a series of simple expressions with || in between, I don't see it as complex enough to need that sort of breaking up

**Comment by levitte** [comment]
> CIs are relevant (I think).

Yup.  I didn't pay attention that some of the guards were `#ifdef` rather than `#ifndef`

File: engines/e_devcrypto.c (hunk 2 above)
Comment (levitte) [reply engines/e_devcrypto.c:1254]: I'm thinking that if there's a day where /dev/crypto supports this properly, we already have the preparation code in place.

**Comment by levitte** [comment]
CIs are much happier now

**Review by t8m — DISMISSED** [review]

**Comment by levitte** [comment]
Fixup added, the logic in `apps/speed.c` wasn't quite right

**Comment by t8m** [comment]
Hmm, still needs some adjustments probably?

**Comment by levitte** [comment]
Yes, those adjustments are in a separate PR, #13706

**Comment by kroeckx** [comment]
speed.c still has a problem

**Comment by levitte** [comment]
> speed.c still has a problem

Unless I missed something, that's fixed by #13706

**Comment by levitte** [comment]
> Ah yup, the changes don't like `no-bulk`, even with #13706 merged in

Fixed, I hope

**Review by t8m — APPROVED** [review]

**Comment by levitte** [comment]
Hmmm, so perhaps this doesn't need to wait on #13706 after all...

**Comment by levitte** [comment]
With #13706, this needed a rebase.  There was one small conflict in `crypto/asn1/i2d_evp.c` which was trivial to clear.

**Comment by t8m** [comment]
Still approved

19. 12. 2020 17:10, 17:10, Richard Levitte <notifications@github.com> napsal/a:

**Comment by levitte** [comment]
Merged

3a1ee3c1993c588a22cb80f1d0eb6237f83a1560 Drop OPENSSL_NO_RSA everywhere

**Pull request closed without merging**